test/integration/compound/compound_test.go

```diff
@@ -0,0 +1,426 @@
+				} else {
+					g.Expect(entry.Status.Targets).To(Equal(quoted(entry.Spec.Text)))
+				}
+				if entry.Spec.TTL != nil {
+					g.Expect(entry.Status.TTL).To(Equal(entry.Spec.TTL))
+				} else {
+					g.Expect(entry.Status.TTL).To(Equal(ptr.To[int64](defaultTTL)))
+				}
+				g.Expect(entry.Status.ObservedGeneration).To(Equal(entry.Generation))
+			}).Should(Succeed())
+		}
+	)
+
+	BeforeEach(func() {
+		//SetDefaultEventuallyTimeout(30 * time.Second)
+
+		ctxLocal := context.Background()
+		ctx0 := ctxutil.CancelContext(ctxutil.WaitGroupContext(context.Background(), "main"))
+		ctx = ctxutil.TickContext(ctx0, controllermanager.DeletionActivity)
+
+		By("Create test Namespace")
+		testNamespace = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "compound-",
+			},
+		}
+		Expect(testClient.Create(ctxLocal, testNamespace)).To(Succeed())
+		log.Info("Created Namespace for test", "namespaceName", testNamespace.Name)
+		testRunID = testNamespace.Name
+
+		DeferCleanup(func() {
+			By("Delete test Namespace")
+			Expect(testClient.Delete(ctxLocal, testNamespace)).To(Or(Succeed(), BeNotFoundError()))
+		})
+
+		By("Start manager")
+
+		go func() {
+			defer GinkgoRecover()
+			args := []string{
+				"--kubeconfig", kubeconfigFile,
+				"--controllers", "dnscontrollers,annotation",
+				"--omit-lease",
+				"--dns-delay", "1s",
+				"--reschedule-delay", "2s",
+				"--lock-status-check-period", "5s",
+				"--pool.size", "2",
+			}
+			runControllerManager(ctx, args)
+		}()
+
+		DeferCleanup(func() {
+			By("Stop manager")
+			if ctx != nil {
+				ctxutil.Cancel(ctx)
+			}
+		})
+
+		mcfg := mock.MockConfig{
+			Name: testRunID,
+			Zones: []mock.MockZone{
+				{ZonePrefix: testRunID + ":first:", DNSName: "first.example.com"},
+				{ZonePrefix: testRunID + ":second:", DNSName: "second.example.com"},
+			},
+		}
+		firstZoneID = mcfg.Zones[0].ZoneID()
+		bytes, err := json.Marshal(&mcfg)
+		Expect(err).NotTo(HaveOccurred())
+
+		providerSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: testRunID,
+				Name:      "mock1-secret",
+			},
+			Data: map[string][]byte{},
+			Type: corev1.SecretTypeOpaque,
+		}
+		Expect(testClient.Create(ctx, providerSecret)).To(Succeed())
+		DeferCleanup(func() {
+			Expect(testClient.Delete(ctx, providerSecret)).To(Succeed())
+		})
+		provider := &v1alpha1.DNSProvider{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: testRunID,
+				Name:      "mock1",
+			},
+			Spec: v1alpha1.DNSProviderSpec{
+				Type:           "mock-inmemory",
+				ProviderConfig: &runtime.RawExtension{Raw: bytes},
+				SecretRef:      &corev1.SecretReference{Name: "mock1-secret", Namespace: testRunID},
+			},
+		}
+		Expect(testClient.Create(ctx, provider)).To(Succeed())
+		DeferCleanup(func() {
+			Expect(testClient.Delete(ctx, provider)).To(Succeed())
+		})
+
+		Eventually(func(g Gomega) {
+			Expect(testClient.Get(ctx, client.ObjectKeyFromObject(provider), provider)).To(Succeed())
```

Comment:
Should this be using the local Gomega instance as well?
```suggestion
			g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(provider), provider)).To(Succeed())
```